Tools/AP_Bootloader/support.cpp

```diff
@@ -364,6 +364,17 @@ void thread_sleep_ms(uint32_t ms)
     }
 }
 
+void thread_sleep_us(uint32_t us)
+{
+    while (us > 0) {
+        // don't sleep more than 65 at a time, to cope with 16 bit
+        // timer
+        const uint32_t dt = us > 6500? 6500: us;
+        chThdSleepMicroseconds(dt);
```

Comment:
might be safer to use chTimeUS2I(1)*us and compare that to (UINT16_MAX-1)? and call chThdSleep on that?

Reply:
notice it is 6.5ms max, and chThdSleepMicroseconds() is what we use elsewhere in the bootloader. I think the call is right